### PR TITLE
Show Warning dialog when click job view node and storage accounts folder node for HDI Reader cluster in Eclipse

### PR DESCRIPTION
**Pull request by t-rufang**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewHDInsightReaderClusterForm.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+		this.defaultStorageRootPath = selectedClusterDetail.getDefaultStorageRootPath();
+	}
+	
+	@Override
+	protected void customizeUI() {
+		this.clusterNameField.setText(selectedClusterDetail.getName());
+		this.clusterNameField.setEditable(false);
+	}
+	
+	private Predicate<IClusterDetail> getSelectedLinkedHdiClusterPredicate(@NotNull String selectedClusterName) {
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewHDInsightReaderClusterForm.java]: I would like suggest to declare a `boolean` return value function, which would be more clear.

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/helpers/HDInsightHelperImpl.java [hunk 2]
```diff
@@ -122,7 +132,67 @@ public static synchronized void initHDInsightLoader() {
...
+                        new AddNewHDInsightReaderClusterForm(PluginUtil.getParentShell(), module, clusterDetail);
+                linkClusterForm.open();
+            }
+        };
+    }
+    
+    public void createRefreshHdiReaderJobsWarningForm(@NotNull HDInsightRootModule module, @NotNull ClusterDetail clusterDetail) {
+		Display.getDefault().asyncExec(() -> {
+			String title = "Cluster Job Access Denied";
+			String warningText = "You have Read-only permission for this cluster. Please ask the cluster owner or user access administrator to upgrade your role to HDInsight Cluster Operator in the Azure Portal, or ‘Link this cluster’ through Ambari credentials to view the corresponding jobs.";
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/helpers/HDInsightHelperImpl.java]: Too long line, just fix it with max 120 chars

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewHDInsightReaderClusterForm.java (hunk 1 above)
Comment (t-rufang) [reply PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewHDInsightReaderClusterForm.java]: In the class `MessageDialog`, the dialog return code is the button id, which is defined by the index of button labels array. In our case, the button labels is `new String[] { "Link this cluster", "Cancel" }`, therefore, the `Link the cluster` button id is 0 while the `Cancel` button id is 1.
```
    @Override
 protected void buttonPressed(int buttonId) {
        setReturnCode(buttonId);
        close();
    }
```

---
In reply to: [285934934](https://github.com/microsoft/azure-tools-for-java/pull/3138#discussion_r285934934) [](ancestors = 285934934)

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewHDInsightReaderClusterForm.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+	}
+	
+	@Override
+	protected void customizeUI() {
+		this.clusterNameField.setText(selectedClusterDetail.getName());
+		this.clusterNameField.setEditable(false);
+	}
+	
+	private boolean getSelectedLinkedHdiCluster(@NotNull IClusterDetail clusterDetail,
+														 @NotNull String selectedClusterName) {
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewHDInsightReaderClusterForm.java]: Whitespace...my fault

**Pull request merged**